crates/jrsonnet-macros/src/lib.rs

```diff
@@ -538,40 +545,79 @@ impl TypedField {
 		})
 	}
 	fn expand_parse(&self) -> TokenStream {
+		if self.is_option {
+			self.expand_parse_optional()
+		} else {
+			self.expand_parse_mandatory()
+		}
+	}
+
+	fn expand_parse_optional(&self) -> TokenStream {
 		let ident = &self.ident;
 		let ty = &self.ty;
+
+		// optional flatten is handled in same way as serde
 		if self.attr.flatten {
-			// optional flatten is handled in same way as serde
-			return if self.is_option {
-				quote! {
-					#ident: <#ty as TypedObj>::parse(&obj).ok(),
-				}
-			} else {
-				quote! {
-					#ident: <#ty as TypedObj>::parse(&obj)?,
-				}
+			return quote! {
+				#ident: <#ty as TypedObj>::parse(&obj).ok(),
 			};
-		};
+		}
 
 		let name = self.name().unwrap();
-		let value = if self.is_option {
-			quote! {
-				if let Some(value) = obj.get(#name.into())? {
-					Some(<#ty as Typed>::from_untyped(value)?)
-				} else {
-					None
-				}
+		let aliases = &self.attr.aliases;
+
+		let value = quote! {
+			if let Some(__value) = obj.get(#name.into())? {
+				Some(<#ty as Typed>::from_untyped(__value)?)
+			} #(else if let Some(__value) = obj.get(#aliases) {
```

Comment:
ObjValue::get returns Result<Option<T>>, but it is being matched as Option<T> here.
It is handled correctly in `expand_parse_mandatory` case

Reply:
I don't understand your comment, sorry. Yes, codegen for Option and non-Option is indeed similar. But I don't know how to unify it without loss of readability.

Let's discuss using an example: 

```rust
#[derive(Typed)]
pub struct KeyValue {
	key: IStr,
	value: Option<Val>,
}
```

Part of codegen:

```rust
fn parse(obj: &ObjValue) -> JrResult<Self> {
	Ok(Self {
		key: if let Some(__value) = obj.get("key".into())? {
			<IStr as Typed>::from_untyped(__value)?
		} else {
			return Err(ErrorKind::NoSuchField("key".into(), vec![]).into());
		},
		value: if let Some(__value) = obj.get("value".into())? {
			Some(<Val as Typed>::from_untyped(__value)?)
		} else {
			None
		},
	})
}
```

How to unify code generation here to unify `expand_parse_optional` and `expand_parse_mandatory`?

As an idea, I could  write `<IStr as Typed>::from_untyped(__value)?.into()` to avoid wrapping with Some, but we already have too many `into` calls.

Reply:
I missed this error. Hmm, thanks! Fixed